renku/core/commands/session.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2022- Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku session commands."""
+
+import webbrowser
+from itertools import chain
+from typing import Optional
+
+from yaspin import yaspin
+
+from renku.core import errors
+from renku.core.commands.format.session import SESSION_FORMATS
+from renku.core.management.command_builder import inject
+from renku.core.management.command_builder.command import Command
+from renku.core.management.interface.client_dispatcher import IClientDispatcher
+from renku.core.models.session import ISessionProvider
+from renku.core.plugins.session import supported_session_providers
+from renku.core.utils import communication
+from renku.core.utils.os import safe_read_yaml
+
+
+@inject.autoparams()
+def _get_renku_project_name(client_dispatcher: IClientDispatcher) -> str:
+    client = client_dispatcher.current_client
+    return f"{client.remote['owner']}/{client.remote['name']}" if client.remote["name"] else f"{client.path.name}"
+
+
+def _safe_get_provider(provider: str) -> ISessionProvider:
+    providers = supported_session_providers()
+    p = next(filter(lambda x: x[1] == provider, providers), None)
+
+    if p is None:
+        raise errors.ParameterError(f"Session provider '{provider}' is not available!")
+    return p[0]
+
+
+def _session_list(config_path: str, format="tabular", provider: Optional[str] = None):
+    project_name = _get_renku_project_name()
+    config = safe_read_yaml(config_path) if config_path else dict()
+
+    providers = supported_session_providers()
+    if provider:
+        providers = list(filter(lambda x: x[1] == provider, providers))
+
+    if len(providers) == 0:
+        raise errors.ParameterError("No session provider is available!")
+
+    sessions_list = list(chain(*map(lambda x: x[0].session_list(config=config, project_name=project_name), providers)))
+    return SESSION_FORMATS[format](sessions_list)
```

Comment:
In the DatasetsApiMixin refactor (and in general) I've started moving this around so the `FORMAT[format](something)` call happens in `renku.cli` not core, since it is rather cli-specific (service wouldn't use tabular format as a return so core shouldn't know about formats, as it's not a generic concept).

E.g. https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/cli/dataset.py#L554 and https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/core/management/dataset/dataset.py#L82

And all the logic should be in `renku.core.management.session` somewhere, this file should only contain the command wrapping functions, e.g. https://github.com/SwissDataScienceCenter/renku-python/blob/develop/renku/core/commands/dataset.py#L44 to separate things better.